### ISSUE TITLE
Index type difference, cannot be detected using the command 'liquibase diff'

### ISSUE THREAD

**Issue by ZHangTong-CHINA** [liquibase/liquibase#6320]
### Search first

- [X] I searched and no similar issues were found

### Description

I found that the official documentation introduces the 'diff' command, and the open source version supports the index type. After practicing and testing, I found that there are problems
![image](https://github.com/user-attachments/assets/a4a52395-5a07-46c9-8744-c46c9e95123c)
![image](https://github.com/user-attachments/assets/58166238-9545-43ce-babb-561d972975f4)
![image](https://github.com/user-attachments/assets/de37abfc-42f5-4008-860b-216b54ab5539)

### Steps To Reproduce

I created two databases locally, one simulating a development environment and the other simulating an integration environment. I performed the following operations:
1.I applied the new index operation in the dev database.
2.I configured the SIT database URL and username password in the liquibase.properties, and then used the 'liquibase diff' command to view the differences between the dev database and the SIT database,I found no difference

### Expected/Desired Behavior

I don't understand why, I hope the official can help me investigate the reason

### Liquibase Version

4.29.2

### Database Vendor & Version

MSYQL 5.7 version installed with Docker

### Liquibase Integration

CLI

### OS and/or Infrastructure Type/Provider

Centos7

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR (Thank you!)

**Comment by ZHangTong-CHINA** [comment]
My 'liquibase. properties' file is as follows
![image](https://github.com/user-attachments/assets/7495c3e7-568b-45cd-ae71-9c28d6274d45)

**Comment by tati-qalified** [comment]
Hi @ZHangTong-CHINA, thank you for reporting this issue.
Unfortunately I haven't been able to replicate it - the missing / unexpected indices show up for me.

One thing I noticed is that, when I run `liquibase diff` between my dev schema and an empty schema, the output contains the `databasechangelog` table differences. Have you executed Liquibase commands in your SIT schema before?

Apart from that, could you run `liquibase snapshot` for either schema, and see if the index shows up in only one of them? [Here](https://docs.liquibase.com/commands/inspection/snapshot.html) is some information on the snapshot command.
Let me know what you find.

Thank you,
Tatiana

**Comment by ZHangTong-CHINA** [comment]
> Hi @ZHangTong-CHINA, thank you for reporting this issue. Unfortunately I haven't been able to replicate it - the missing / unexpected indices show up for me.
> 
> One thing I noticed is that, when I run `liquibase diff` between my dev schema and an empty schema, the output contains the `databasechangelog` table differences. Have you executed Liquibase commands in your SIT schema before?
> 
> Apart from that, could you run `liquibase snapshot` for either schema, and see if the index shows up in only one of them? [Here](https://docs.liquibase.com/commands/inspection/snapshot.html) is some information on the snapshot command. Let me know what you find.
> 
> Thank you, Tatiana

I checked my dev and sit environments using the command 'liquibase snapshot' and did not find any differences. But I may have found the cause of the problem. I have defined different types of indexes for the same field in the same table. Is this the problem?
![image](https://github.com/user-attachments/assets/8b73822f-879f-4550-b415-185223d990b3)

**Comment by tati-qalified** [comment]
@ZHangTong-CHINA You're right, it seems like Liquibase is expecting only one index for any specific column.

I tried having two indexes on the same column and running `liquibase snapshot`, and here is the relevant part of the output:
```
indexes: 
    idxName2
        columns: 
            firstname
        unique: false
```

Another index called `idxName1` should have been found.

Would you be willing to submit a PR with a fix for this bug? Our development team would be available to provide guidance and answer any questions you may have.
If not, we'll be leaving this issue open for the community to propose a fix.

Thank you,
Tatiana

**Comment by ZHangTong-CHINA** [comment]
@tati-qalified Thank you, I discovered this issue during my learning process, but whether this approach is standardized and reasonable requires professional evaluation to determine if it needs to be fixed. I am not a developer and cannot submit PR.

**Comment by rberezen** [comment]
Re-triage: confirmed still reproducible on current `main` (5.x), MySQL 8. Two indexes on the same column:
```sql
CREATE INDEX idxName1 ON person(firstname);
CREATE INDEX idxName2 ON person(firstname);
```
`liquibase snapshot` captures only `idxName1` — `idxName2` appears **0 times** in the snapshot. Consequence for `diff`: a reference DB with both indexes vs a target with only `idxName1` reports `Missing Index(s): NONE`, so the missing second index is invisible (exactly as @tati-qalified narrowed it in 2024).

Root cause is snapshot-level: indexes are keyed/deduped by their column set, so multiple indexes on identical columns collapse to one. Keeping open.
